controller/src/main/java/io/pravega/controller/store/PravegaTablesScope.java

```diff
@@ -88,57 +91,66 @@ public String getName() {
-                                                                                                .thenAccept(v -> {
-                                                                                                    log.debug(context.getRequestId(), "table for kvts created {}", kvtsTableName);
-                                                                                                    if (e != null) {
-                                                                                                        throw new CompletionException(e);
-                                                                                                    }
-                                                                                                })),
-                                               getReaderGroupsInScopeTableName(context)
-                                                       .thenCompose(rgTableName -> storeHelper.createTable(rgTableName, context.getRequestId())
-                                                                                              .thenAccept(v -> {
-                                                                                                  log.debug(context.getRequestId(), "table for reader groups created {}", rgTableName);
-                                                                                                  if (e != null) {
-                                                                                                      throw new CompletionException(e);
-                                                                                                  }
-                                                                                              })),
-                                               getAllStreamTagsInScopeTableNames(context)
-                                                       .thenCompose(tagTableNames -> {
-                                                           List<CompletableFuture<Void>> futures = new ArrayList<>();
-                                                           for (String tableName : tagTableNames) {
-                                                               futures.add(storeHelper.createTable(tableName, context.getRequestId()));
-                                                           }
-                                                           return Futures.allOf(futures);
-                                                       })
-                                                       .thenAccept(v -> {
-                                                           if (e != null) {
-                                                               throw new CompletionException(e);
-                                                           }
-                                                       }));
+        return checkScopeExistsInTable(scopeName, context).thenCompose(exist -> {
```

Comment:
The check for scope to exist in _deletingScopes table should be done inside `AbstractStreamMetadataStore.createScope()` before invoking this createScope() in this class. This code can stay as-is.

Reply:
We do not have required APIs to access the internal things in `AbstractStreamMetadataStore.java` for this